From e13bcba30bf74d952e32b5810a0b59abb13d832d Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Tue, 19 Nov 2024 10:10:37 +0100
Subject: [PATCH 1298/1695] drivers: stm32_tamp: fix switched masks when
 applying RIF configuration

Fix switched masks when applying RIF configuration between the
privilege access mask and the secure level access mask.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git cf27cc93e997d65945bca3547bf48f6232c59d2f drivers: stm32_tamp: fix switched masks when applying RIF configuration]
Change-Id: I45cf91dadb377ce71a98ba6545e107ec607afb9c
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/418037
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/drivers/stm32_tamp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/drivers/stm32_tamp.c b/core/drivers/stm32_tamp.c
index 0a2298a3f..72036bce5 100644
--- a/core/drivers/stm32_tamp.c
+++ b/core/drivers/stm32_tamp.c
@@ -588,9 +588,9 @@ static void apply_rif_config(void)
 
 	/* Security and privilege RIF configuration */
 	io_clrsetbits32(base + _TAMP_PRIVCFGR,
-			_TAMP_PRIVCFGR_MASK & access_mask_sec_reg, privcfgr);
+			_TAMP_PRIVCFGR_MASK & access_mask_priv_reg, privcfgr);
 	io_clrsetbits32(base + _TAMP_SECCFGR,
-			_TAMP_SECCFGR_BUT_BKP_MASK & access_mask_priv_reg,
+			_TAMP_SECCFGR_BUT_BKP_MASK & access_mask_sec_reg,
 			seccfgr);
 
 	if (!stm32_tamp.pdata.is_tdcid)
-- 
2.47.2

